test/unit/db.test.ts

```diff
@@ -0,0 +1,50 @@
+import { expect } from 'chai';
+
+import { MongoClient } from '../../src';
+import { Db, DbOptions } from '../../src/db';
+import { ReadPreference } from '../../src/read_preference';
+
+describe('Db - test secondaryOk matches readPreference', function () {
```

Comment:
Just a few points of feedback on the tests:

1) Can we call this describe something like "Db" or "Db instance", and then have all these tests nested in a second describe block "secondaryOk"?

2) Let's make sure we don't keep the .only on the tests. @nbbeeken  recently added a lint rule for this, but it might not be in your branch yet (could be worth a rebase)

3) Let's update the expect assertions to `expect(mydb).property('secondaryOk').to.be.<true/false>` and ```expect(mydb).property(`${legacy_secondary_ok}\`).to.be.<true/false>```